### ISSUE TITLE
BUG / DOC: DataFrame.value_counts doesn't sort the result, docstring says it does

### ISSUE THREAD

**Issue by TomAugspurger** [ContinuumIO/dask#442]
This is dask `0.6.0`.

```python
In [8]: import dask.dataframe as dd

In [9]: df = dd.read_csv('tmp.csv')

In [10]: df.head()
Out[10]:
   A  B
0  a  1
1  b  2
2  a  1
3  a  1
4  b  2
5  b  2
6  b  2
7  c  3
8  b  2
9  c  3

In [11]: df.A.value_counts().compute()
Out[11]:
a    3
b    5
c    2
dtype: int64
```

I was surprised when it *did* say the results would be sorted (by default).

```
Returns object containing counts of unique values.

The resulting object will be in descending order so that the
first element is the most frequently-occurring element.
Excludes NA values by default.

Parameters
----------
normalize : boolean, default False
    If True then the object returned will contain the relative
    frequencies of the unique values.
sort : boolean, default True
    Sort by values
ascending : boolean, default False
    Sort in ascending order
```

Not sure if this is a problem with the docstring or the implementation. A couple other things that may help track this down. It seems to be sorting the index (same thing on a larger example), and it raises a `TypeError` when you pass in sort.

```python
In [23]: df.A.value_counts(sort=True).compute()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-23-9f674ed286e0> in <module>()
----> 1 df.A.value_counts(sort=True).compute()

TypeError: value_counts() got an unexpected keyword argument 'sort'
````

**Comment by mrocklin** [comment]
We just stole the docstring from Pandas.  I wasn't personally aware of this aspect of `value_counts` though and the tests were blind to it.  Community feedback/review like this is really valuable on things like this.

Fixed in #443

**Comment by TomAugspurger** [comment]
I was confused when `dd.A.value_counts?` actually pointed back to `pandas/core/base.py`. Makes sense now.

I'll keep toying with `dask.dataframe` :+1:
